lib/services/parser.go

```diff
@@ -53,6 +53,93 @@ var (
 	CertAuthorityTypeExpr = builder.Identifier(`system.catype()`)
 )
 
+// PredicateAllEndWith is a custom function to test if a string ends with a
+// particular suffix. If given a `[]string` as the first argument, all values
+// must have the given suffix (2nd argument).
+func PredicateAllEndWith(a interface{}, b interface{}) predicate.BoolPredicate {
+	return func() bool {
+		// bval is the suffix and must always be a plain string.
+		bval, ok := b.(string)
+		if !ok {
+			return false
+		}
+
+		switch aval := a.(type) {
+		case string:
+			return strings.HasSuffix(aval, bval)
+		case []string:
+			for _, val := range aval {
+				if !strings.HasSuffix(val, bval) {
+					return false
+				}
+			}
+			return true
+		default:
+			return false
+		}
+	}
+}
+
+// PredicateAllEqual is a custom function to test if all entries in a []string
+// are equal to a certain value. This is primarily useful for comparing string
+// fields that are only expected to contain a single, specific value.
+func PredicateAllEqual(a interface{}, b interface{}) predicate.BoolPredicate {
+	return func() bool {
+		// bval is the suffix and must always be a plain string.
+		bval, ok := b.(string)
+		if !ok {
+			return false
+		}
+
+		switch aval := a.(type) {
+		case string:
+			return aval == bval
+		case []string:
+			for _, val := range aval {
+				if val != bval {
+					return false
+				}
+			}
+			return true
+		default:
+			return false
+		}
+	}
+}
+
+// PredicateIsSubset determines if the first parameter is contained within the
+// variadic args. Only strings are supported; the first parameter may either
```

Comment:
This comment could be a bit clearer, it says "only strings are supported" followed by "may either be a string or list of strings" - Maybe include something that suggests that the variadic args may only be strings.

Reply:
That's fair, I'll update the comment.